common/core/src/authentication_provider.ts

```diff
@@ -26,4 +26,5 @@ export interface AuthenticationProvider {
   type: AuthenticationType;
   getDeviceCredentials(callback: Callback<TransportConfig>): void;
   getDeviceCredentials(): Promise<TransportConfig>;
-}
+  setTokenRenewalValues?(tokenValidTimeInSeconds: number, tokenRenewalMarginInSeconds: number): void;
```

Comment:
>tokenRenewalMarginInSeconds [](start = 58, length = 27)

Nitpick: I don't really like the name "Margin" for this. I think simple "tokenRenewalTimeInSeconds" would do it.

Reply:
It mimics the name used in the constructor.